packages/indexer/native/src/lib.rs

```diff
@@ -430,32 +432,44 @@ declare_types! {
                 return cx.throw_error(format!("Error assembling script: {:?}", script.unwrap_err()));
             }
             let script = script.unwrap();
-            let prefix = if cx.argument::<JsNumber>(2)?.value() as u32 == 1 {
+            let script_type = cx.argument::<JsNumber>(2)?.value() as u32;
```

Comment:
I don't see `script_type` getting used below, why creating a variable?

Reply:
It's used one line below the variable declaration, to deciding the `prefix`.

Reply:
Yes but this is the only place we are using `script_type` here, I was curious why embedding it in `prefix` calculation won't work? What's the rationale behind splitting it to a varaible?

Reply:
Get it. I'll merge them into one line in the next commit.